src/test/app/DepositAuth_test.cpp

```diff
@@ -724,14 +807,760 @@ struct DepositPreauth_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testCredentialsPayment()
+    {
+        using namespace jtx;
+
+        const char credType[] = "abcde";
+        Account const issuer{"issuer"};
+        Account const alice{"alice"};
+        Account const bob{"bob"};
+        Account const maria{"maria"};
+        Account const john{"john"};
+
+        {
+            testcase("Payment failed with disabled credentials rule.");
+
+            Env env(*this, supported_amendments() - featureCredentials);
+
+            env.fund(XRP(5000), issuer, bob, alice);
+            env.close();
+
+            // Bob require preauthorization
+            env(fset(bob, asfDepositAuth));
+            env.close();
+
+            // Setup DepositPreauth object failed - amendent is not supported
+            env(deposit::authCredentials(bob, {{issuer, credType}}),
+                ter(temDISABLED));
+            env.close();
+
+            // But can create old DepositPreauth
+            env(deposit::auth(bob, alice));
+            env.close();
+            // And alice can pay without credentials
+            env(pay(alice, bob, XRP(10)));
+            env.close();
+
+            // And alice can pay with any credentials, as they aren't checked if
+            // amendement is not enabled
+            std::string const invalidIdx =
+                "0E0B04ED60588A758B67E21FBBE95AC5A63598BA951761DC0EC9C08D7E"
+                "01E034";
+            env(pay(alice, bob, XRP(10)), credentials::ids({invalidIdx}));
```

Comment:
Providing the `CredentialIDs` field should also result in `temDISABLED`.

Reply:
fixed